internal/provider/resources/resource_hybrid_cluster_workspace_authorization.go

```diff
@@ -0,0 +1,357 @@
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	// Wait for the cluster to be updated (or fail)
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{string(platform.ClusterStatusCREATING), string(platform.ClusterStatusUPDATING)},
+		Target:     []string{string(platform.ClusterStatusCREATED), string(platform.ClusterStatusUPDATEFAILED), string(platform.ClusterStatusCREATEFAILED)},
+		Refresh:    ResourceRefreshFunc(ctx, r.platformClient, r.organizationId, cluster.JSON200.Id),
+		Timeout:    1 * time.Hour,
+		MinTimeout: 1 * time.Minute,
+	}
+
+	// readyCluster is the final state of the cluster after it has reached a target status
+	readyCluster, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("Hybrid cluster authorization creation", err.Error())
+		return
+	}
+
+	diags = data.ReadFromResponse(readyCluster.(*platform.Cluster))
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("Created hybrid cluster workspace authorization for cluster: %v", data.ClusterId.ValueString()))
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *hybridClusterWorkspaceAuthorizationResource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var data models.HybridClusterWorkspaceAuthorizationResource
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cluster, err := r.platformClient.GetClusterWithResponse(ctx, r.organizationId, data.ClusterId.ValueString())
+	if err != nil {
+		tflog.Error(ctx, "failed to get cluster", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get cluster, got error: %s", err))
+		return
+	}
+	statusCode, diagnostic := clients.NormalizeAPIError(ctx, cluster.HTTPResponse, cluster.Body)
+	// If the resource no longer exists, it is recommended to ignore the errors
+	// and call RemoveResource to remove the resource from the state. The next Terraform plan will recreate the resource.
+	if statusCode == http.StatusNotFound {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	diags := data.ReadFromResponse(cluster.JSON200)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("Read cluster workspace authorization for: %v", data.ClusterId.ValueString()))
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *hybridClusterWorkspaceAuthorizationResource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var data models.HybridClusterWorkspaceAuthorizationResource
```

Comment:
since update and create are so similar, can we do something similar like this
https://github.com/astronomer/terraform-provider-astro/blob/a5f22de4495f3481537d19fc9a84cf0f07d2d503/internal/provider/resources/resource_team_roles.go#L77